packages/truffle-deployer/legacy.js

```diff
@@ -0,0 +1,81 @@
+var expect = require("truffle-expect");
+var DeferredChain = require("./src/deferredchain");
+var deploy = require("./src/actions/deploy");
+var deployMany = require("./src/actions/deploymany");
+var link = require("./src/actions/link");
+var create = require("./src/actions/new");
+
+function LegacyDeployer(options = {}) {
+  var self = this;
+
+  expect.options(options, ["provider", "network", "network_id"]);
+
+  this.chain = new DeferredChain();
+  this.logger = options.logger || { log: function() {} };
+  this.known_contracts = {};
+  (options.contracts || []).forEach(function(contract) {
+    self.known_contracts[contract.contract_name] = contract;
+  });
+  this.network = options.network;
+  this.network_id = options.network_id;
+  this.provider = options.provider;
+  this.basePath = options.basePath || process.cwd();
+}
+
+// Note: In all code below we overwrite this.chain every time .then() is used
+// in order to ensure proper error processing.
+
+LegacyDeployer.prototype.start = function() {
+  return this.chain.start();
+};
+
+LegacyDeployer.prototype.link = function(library, destinations) {
+  return this.queueOrExec(link(library, destinations, this)); //legacyLink(library, destinations, this));
+};
+
+LegacyDeployer.prototype.deploy = function() {
+  var args = Array.prototype.slice.call(arguments);
+  var contract = args.shift();
+
+  if (Array.isArray(contract)) {
+    return this.queueOrExec(deployMany(contract, this));
+  } else {
+    return this.queueOrExec(deploy(contract, args, this));
+  }
```

Comment:
Is the only substantive difference here that it uses `deploy` / `deployMany` directly instead of inheriting `this.executeDeployment` from the `Deployment` base class?

If so, maybe we can do away with this file.

Reply:
iirc, the `Deployment` base class starts doing web3 1.0-specific things behind the scenes which causes legacy deployment to break. I tried a number of approaches to not have to pull in the main legacy truffle-deployer file, but never got it to work.